From 9db232aaf07d9ab4a6c15bbcd643a9a2b674bc3c Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Mon, 26 Aug 2019 14:45:01 +0200
Subject: [PATCH] send error response on too large results

so far livestatus would quietly end a query if it hits the maximum output size without
the client being able to notice. Since it is better to get no data and an error instead
of wrong or incomplete data and not even knowing it we better bail out with an error.

Signed-off-by: Sven Nierlein <sven@nierlein.de>
---
 src/Query.cc | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/Query.cc b/src/Query.cc
index 1a84c73..4bde438 100644
--- a/src/Query.cc
+++ b/src/Query.cc
@@ -914,9 +914,7 @@ bool Query::processDataset(void *data)
 {
     if (_output->size() > g_max_response_size) {
         logger(LG_INFO, "Maximum response size of %d bytes exceeded!", g_max_response_size);
-        // _output->setError(RESPONSE_CODE_LIMIT_EXCEEDED, "Maximum response size of %d reached", g_max_response_size);
-        // currently we only log an error into the log file and do
-        // not abort the query. We handle it like Limit:
+        _output->setError(RESPONSE_CODE_LIMIT_EXCEEDED, "Maximum response size of %d reached", g_max_response_size);
         return false;
     }
 
